### ISSUE TITLE
repeated headers?

### ISSUE THREAD

**Issue by lucaszanella** [sdroege/rtsp-types#6]
I'm implementing digest authentication on my RTSP client. On https://tools.ietf.org/html/rfc7616#page-18, it has this example:

```
HTTP/1.1 401 Unauthorized
   WWW-Authenticate: Digest
       realm="http-auth@example.org",
       qop="auth, auth-int",
       algorithm=SHA-256,
       nonce="7ypf/xlj9XXwfDPEoM4URrv/xwf94BcCAzFZH4GiTo0v",
       opaque="FQhe/qaU925kfnzjCev0ciny7QMkPqMAFRtzCUYo5tdS"
   WWW-Authenticate: Digest
       realm="http-auth@example.org",
       qop="auth, auth-int",
       algorithm=MD5,
       nonce="7ypf/xlj9XXwfDPEoM4URrv/xwf94BcCAzFZH4GiTo0v",
       opaque="FQhe/qaU925kfnzjCev0ciny7QMkPqMAFRtzCUYo5tdS"

```

even though this RFC is not from RTSP 1.0, but an improvement of the one RTSP 1.0 uses, I'm using it because the old one only implements MD5.

See that it permits repeated header names. However, on this implementation, repeated header names are not possible because it's a `BTreeMap`. I couldn't find anything in RTSP 1.0 that said no repeated headers are prohibited.

What you think about this?

**Comment by sdroege** [comment]
See https://tools.ietf.org/html/rfc7826#section-5.2
> Multiple header fields with the same field-name MAY be present in a
   message if and only if the entire field-value for that header field
   is defined as a comma-separated list.  It MUST be possible to combine
   the multiple header fields into one "field-name: field-value" pair,
   without changing the semantics of the message, by appending each
   subsequent field-value to the first, each separated by a comma.  The
   order in which header fields with the same field-name are received is
   therefore significant to the interpretation of the combined field
   value; thus, a proxy MUST NOT change the order of these field-values
   when a message is forwarded.

And https://tools.ietf.org/html/rfc7826#section-18.58
> The field-value
   consists of **at least one** challenge that indicates the authentication
   scheme(s) and parameters applicable to the Request-URI.  This header
   MUST only be used in response messages related to client to server
   requests.

And then of course https://tools.ietf.org/html/rfc7235.

In short: it's a comma separated list of one or more challenges and your example above is equivalent to

```
HTTP/1.1 401 Unauthorized
   WWW-Authenticate: Digest
       realm="http-auth@example.org",
       qop="auth, auth-int",
       algorithm=SHA-256,
       nonce="7ypf/xlj9XXwfDPEoM4URrv/xwf94BcCAzFZH4GiTo0v",
       opaque="FQhe/qaU925kfnzjCev0ciny7QMkPqMAFRtzCUYo5tdS",
       Digest
       realm="http-auth@example.org",
       qop="auth, auth-int",
       algorithm=MD5,
       nonce="7ypf/xlj9XXwfDPEoM4URrv/xwf94BcCAzFZH4GiTo0v",
       opaque="FQhe/qaU925kfnzjCev0ciny7QMkPqMAFRtzCUYo5tdS"
```

The parser is currently collecting multiple headers with the same name in this way.

**Comment by lucaszanella** [comment]
Thank you. I was reading the old one (rfc2326) which does not have this requirement, as far as I could tell. I know 2.0 is not backwards compatible but my client is working on just the subset of yours that works on 1.0.
